### PR TITLE
feat: Add LeakCanary for debug implementation

### LINKED ISSUE #1750: Cannot access settings and other options without authentication
**Describe the bug**
Split other options with the profile.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### LINKED ISSUE #1752: Add Leak Canary to detect memory leak in the app
**Describe the feature you'd like**
Add LeakCanary android library to detect memory leaks.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by liveHarshit**
Fixes #1752

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -184,6 +184,9 @@ dependencies {
     releaseImplementation 'com.github.iamareebjamal:stetho-noop:1.2.1'
     testImplementation 'com.github.iamareebjamal:stetho-noop:1.2.1'
 
+    //LeakCanery
```
Comment (ShridharGoel) [inline app/build.gradle]: LeakCanary*

**Comment by liveHarshit** [comment]
@iamareebjamal Leak analysis failed while the app is crashing due to memory leak issue #1705 
Logs - 
```

java.lang.OutOfMemoryError: Failed to allocate a 2000016 byte allocation with 406368 free bytes and 396KB until OOM
at leakcanary.internal.LongToIntSparseArray.<init>(LongToIntSparseArray.kt:20)
at leakcanary.HprofParser.<init>(HprofParser.kt:106)
at leakcanary.HprofParser.<init>(HprofParser.kt:73)
at leakcanary.HprofParser$Companion.open(HprofParser.kt:908)
at leakcanary.HeapAnalyzer.checkForLeaks(HeapAnalyzer.kt:97)
at leakcanary.internal.HeapAnalyzerService.onHandleIntentInForeground(HeapAnalyzerService.kt:41)
at leakcanary.internal.ForegroundService.onHandleIntent(ForegroundService.kt:56)
at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:68)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:163)
at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by liveHarshit** [comment]
Please merge it.

**Comment by ShridharGoel** [comment]
Isn't it required to configure LeakCanary?

https://github.com/fossasia/open-event-orga-app/blob/e17570c418f51a63e7b9ec2ec01bd3ffd8d6f2b4/app/src/main/java/com/eventyay/organizer/OrgaApplication.java#L120-L127

**Comment by liveHarshit** [comment]
> Isn't it required to configure LeakCanary?
> 
> https://github.com/fossasia/open-event-orga-app/blob/e17570c418f51a63e7b9ec2ec01bd3ffd8d6f2b4/app/src/main/java/com/eventyay/organizer/OrgaApplication.java#L120-L127

It's for an older version.

**Comment by liveHarshit** [comment]
@ShridharGoel You can update LeakCanary in orga app also with a newer release.

**Pull request merged**